core/src/test/java/org/everit/json/schema/NumberSchemaTest.java

```diff
@@ -253,5 +258,26 @@ public void requiresInteger_nonNullable() {
                 .input(JSONObject.NULL)
                 .expect();
     }
+    
+    @Test
+    public void accepts_bigInteger() {
+        NumberSchema subject = NumberSchema.builder().requiresInteger(true).build();
+        
+        tryIntegerTypes(subject);
+    }
+    
+    @Test
+    public void accepts_bigInteger_noRequire() {
+        NumberSchema subject = NumberSchema.builder().build();
+        
+        tryIntegerTypes(subject);
+    }
+
+    private void tryIntegerTypes(NumberSchema subject) {
+        subject.validate(BigInteger.valueOf(123123123123123L));
+        subject.validate(BigDecimal.valueOf(123123123123123L));
+        subject.validate(new AtomicInteger(123));
+        subject.validate(new AtomicLong(123L));
```

Comment:
It would make sense to differentiate between integral types (`BigInteger`, `AtomicInteger` and `AtomicLong`), and number types (`BigDecimal`). JSON schema has both "number" and "integer" keywords, and these are both handles by `NumberSchema`. `BigDecimal`  is definitely a number but often not an integer.

Reply:
my bad, BigDecimal is in the wrong place